exporter/awsemfexporter/config.go

```diff
@@ -53,10 +54,10 @@ type Config struct {
 	ParseJSONEncodedAttributeValues []string `mapstructure:"parse_json_encoded_attr_values"`
 
 	// MetricDeclarations is the list of rules to be used to set dimensions for exported metrics.
-	MetricDeclarations []*MetricDeclaration `mapstructure:"metric_declarations"`
+	MetricDeclarations []*cwmetrics.MetricDeclaration `mapstructure:"metric_declarations"`
```

Comment:
A public struct should only include public fields I think these config ones need to be moved back from internal

Reply:
Thanks, I want to learn what's the downside to keep these config structs in internal?  
These config structs also have methods defined on it. I have to move the related file (eg, `metric_delaration.go`) out from internal to prevent import cycles if the structs are moved out.

Reply:
Would the struct field be usable while it's in internal from another project? Generally we wouldn't want a user to run into a confusing situation where they have access to some public object but some fields on it don't work. Either the field is private (lowercase `metricDeclarations`) or it's public and the type should be available to the user to reflect that I guess.

Reply:
These structs won't be used by the other projects. It is also the main reason why I moved it into `awsemfexp/internal` so it is only public visible inside `emfexporter` module. 

I think I have 2 options here.
1. remove `awsemfexp/internal` folder and rename the structs and APIs not want to be exposed to lowercase in `emfexporter` package
2. Keep `awsemfexp/internal` folder and update all config structs fields to uppercase so it will be all visible to `emfexporter` package.